kairon/shared/actions/utils.py

```diff
@@ -73,6 +73,16 @@ def execute_http_request(http_url: str, request_method: str, request_body=None,
 
         return http_response_as_json
 
+    @staticmethod
+    def encrypt_secrets(request_body: dict, tracker_data: dict):
```

Comment:
this logic does not cover the case when my json is nested which will be the case most of the time because that is the feature. Find out a way where you are able to find if a value from key vault is used in nested json. Also, the logic is not right. You are searching for key but infact you should be searching for value